lua/nvim-navic/init.lua

```diff
@@ -25,6 +25,106 @@ local function request_symbol(for_buf, handler, client_id)
 	)
 end
 
+-- relation of 'other' with repect to 'symbol'
+local function symbol_relation(symbol, other)
+	local s = symbol.scope
+	local o = other.scope
+
+	if
+		o["end"].line < s["start"].line
+		or (o["end"].line == s["start"].line and o["end"].character < s["start"].character)
+	then
+		return "before"
+	end
+
+	if
+		o["start"].line > s["end"].line
+		or (o["start"].line == s["end"].line and o["start"].character > s["end"].character)
+	then
+		return "after"
+	end
+
+	if
+		(
+			o["start"].line < s["start"].line
+			or (o["start"].line == s["start"].line and o["start"].character <= s["start"].character)
+		)
+		and (
+			o["end"].line > s["end"].line
+			or (o["end"].line == s["end"].line and o["end"].character >= s["end"].character)
+		)
+	then
+		return "around"
+	end
+
+	return "within"
+end
+
+-- Construct tree structure based on scope information
+-- Could be inaccurate ?? Not intended to be used like this...
```

Comment:
> Not intended to be used like this

This is a bit ambiguous -- maybe "Scope information isn't intended to be used like this"?

Reply:
Yep.. will make that clearer.